src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -139,5 +139,98 @@ public static async Task TestBOMWithNoJsonValue()
                     async () => await JsonSerializer.DeserializeAsync<byte>(stream));
             }
         }
+
+        public static IEnumerable<object[]> BOMWithJsonElementTestData
+        {
+            get
+            {
+                yield return CreateStreamArgs(100);
+                yield return CreateStreamArgs(200);
+                yield return CreateStreamArgs(400);
+                yield return CreateStreamArgs(800);
+                yield return CreateStreamArgs(1600);
+
+                static object[] CreateStreamArgs(int count)
+                {
+                    byte[] objBytes = Encoding.UTF8.GetBytes(
+                        @"{""Test"":{},""Test2"":[],""Test3"":{""Value"":{}},""PersonType"":0,""Id"":2}");
+
+                    byte[] utf8Bom = Encoding.UTF8.GetPreamble();
+                    byte[] comma = Encoding.UTF8.GetBytes(@",");
+                    byte[] startArray = Encoding.UTF8.GetBytes(@"[");
+                    byte[] endArray = Encoding.UTF8.GetBytes(@"]");
+
+                    var stream = new MemoryStream();
+
+                    stream.Write(utf8Bom, 0, utf8Bom.Length);
+                    stream.Write(startArray, 0, startArray.Length);
+
+                    for (int i = 1; i <= count; i++)
+                    {
+                        stream.Write(objBytes, 0, objBytes.Length);
+
+                        if (i < count)
+                        {
+                            stream.Write(comma, 0, comma.Length);
+                        }
+                    }
+
+                    stream.Write(endArray, 0, endArray.Length);
+
+                    stream.Position = 0;
+                    return new object[] { stream, count };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(BOMWithJsonElementTestData))]
+        public static async Task TestBOMWithShortAndLongBuffers(Stream stream, int count)
+        {
+            JsonElement[] value;
+
+            // Try with 16K. This should cause copy of the buffer for remaining bytes.
+            JsonSerializerOptions options16K = new JsonSerializerOptions()
```

Comment:
nit: Move different default buffer size into an input, and try a few other values (given the BOM length is 3, maybe: 1, 2, 3, 4, 4k, 8k, 16k, 100k).

Reply:
The test method is already using [MemberData] to obtain the varying size Streams, so we can't add another one. Also, the test already is fairly slow (.5 seconds in Debug build) so unless there's a real case here I'd prefer not to make it twice as slow etc. 

Also I verified the two code branches that deal with the "start" variable for the BOM are hit with the existing tests.

Reply:
I did refactor a bit more to more easily add additional buffer sizes, but as mentioned earlier I don't see a reason at this time for values other than 1 byte and 16K.